### PR TITLE
Add channel#errorReason

### PR DESCRIPTION
**Pull request by SimonWoolf**

### CODE REVIEW

File: spec/realtime/channel.test.js [hunk 1]
```diff
@@ -168,10 +168,13 @@ define(['ably', 'shared_helper', 'async'], function(Ably, helper, async) {
...
 			var realtime = helper.AblyRealtime(realtimeOpts);
 			realtime.connection.once('connected', function() {
-				realtime.channels.get(':hell').attach(function(err) {
+				var channel = realtime.channels.get(':hell')
+				channel.attach(function(err) {
 					if(err) {
-						test.expect(2);
+						test.expect(4);
 						test.ok(true, 'Attach failed as expected');
+						test.equal(channel.errorReason.code, 40010, 'Attach error was set as the channel errorreason');
```
Comment (paddybyers) [inline spec/realtime/channel.test.js]: trivial s/errorreason/errorReason

**Comment by paddybyers** [comment]
LGTM

**Comment by mattheworiordan** [comment]
LGTM 👍

**Comment by paddybyers** [comment]
Closing in favour of https://github.com/ably/ably-js/issues/269

**Comment by SimonWoolf** [comment]
Merged as a5ad0b2

**Pull request closed without merging**